Aliases provided in a `<link>` tag for a Svelte 3 file aren't resolved by known build-system aliases

<!--- Provide a general summary of the issue in the Title above -->
O' LORD OF MODULAR CSS. 

I'm running into an issue where I'd like to use the aliases that my current rollup configuration knows about to point to a specific `css` file to use in a given Svelte3 component. 

like so:
`<link rel="stylesheet" href="shared/styles/target.css">` 

Normally the way I structure the css is by referencing a stylesheet in the same general directory as the component, such that referencing is as simple as 
`<link rel="stylesheet" href="./target.css">`

However, again, I'm running into the issue when I try to use aliases that my rollup configuration knows about. 

## Expected Behavior
<!--- If you're describing a bug, tell us what should happen -->
<!--- If you're suggesting a change/improvement, tell us how it should work -->
`<link rel="stylesheet" href="shared/styles/target.css">` is resolved properly to point to the correct file, where `shared/` is a convenient alias pointing to another folder in my project. 

## Current Behavior
<!--- If describing a bug, tell us what happens instead of the expected behavior -->
<!--- If suggesting a change/improvement, explain the difference from current behavior -->
`<link rel="stylesheet" href="shared/styles/target.css">`  causes an error :

`[!] (plugin svelte) Error: Cannot find module 'shared/styles/target.css'`

## Possible Solution
<!--- Not obligatory, but suggest a fix/reason for the bug, -->
<!--- or ideas how to implement the addition or change -->
The resolution of a given `href` should probably be injected with known build time aliases (or at least the resolver should expose an option to provide it with a pairing of aliases to paths to resolve against)

--- a/package.json
+++ b/package.json
@@ -18,7 +18,6 @@
     "prepublishOnly": "npm run parsers",
     "pretest": "npm run parsers",
     "test": "jest",
-    "posttest": "npm run lint",
     "watch": "jest --watch",
     "www:build": "lerna run --stream build packages/www",
     "www:start": "lerna run --stream start packages/www",

--- a/packages/processor/processor.js
+++ b/packages/processor/processor.js
@@ -9,24 +9,25 @@ const postcss   = require("postcss");
 const slug      = require("unique-slug");
 const mapValues = require("lodash/mapValues");
 
-const output    = require("./lib/output.js");
-const message   = require("./lib/message.js");
-const relative  = require("./lib/relative.js");
-const tiered    = require("./lib/graph-tiers.js");
-const resolve   = require("./lib/resolve.js");
-const normalize = require("./lib/normalize.js");
+const output        = require("./lib/output.js");
+const message       = require("./lib/message.js");
+const relative      = require("./lib/relative.js");
+const tiered        = require("./lib/graph-tiers.js");
+const normalize     = require("./lib/normalize.js");
+const { resolvers } = require("./lib/resolve.js");
 
 const noop = () => true;
 
-const params = ({ _options, _files, _graph, _resolve }, args) => Object.assign(
+const params = ({ _options, _files, _graph, resolve }, args) => Object.assign(
     Object.create(null),
     _options,
     _options.postcss,
     {
-        from    : null,
-        files   : _files,
-        graph   : _graph,
-        resolve : _resolve,
+        resolve,
+        
+        from  : null,
+        files : _files,
+        graph : _graph,
     },
     args
 );
@@ -70,7 +71,8 @@ class Processor {
             // eslint-disable-next-line no-empty-function
             () => {};
 
-        this._resolve = resolve.resolvers(options.resolvers);
+        // Expose resolving API publicly, because it's useful
+        this.resolve = resolvers(options.resolvers);
 
         this._normalize = normalize.bind(null, this._options.cwd);
 

--- a/packages/processor/processor.js
+++ b/packages/processor/processor.js
@@ -18,16 +18,15 @@ const { resolvers } = require("./lib/resolve.js");
 
 const noop = () => true;
 
-const params = ({ _options, _files, _graph, resolve }, args) => Object.assign(
+const params = ({ _options, _files, _graph, _resolve }, args) => Object.assign(
     Object.create(null),
     _options,
     _options.postcss,
     {
-        resolve,
-        
-        from  : null,
-        files : _files,
-        graph : _graph,
+        from    : null,
+        files   : _files,
+        graph   : _graph,
+        resolve : _resolve,
     },
     args
 );
@@ -71,9 +70,7 @@ class Processor {
             // eslint-disable-next-line no-empty-function
             () => {};
 
-        // Expose resolving API publicly, because it's useful
-        this.resolve = resolvers(options.resolvers);
-
+        this._resolve = resolvers(options.resolvers);
         this._normalize = normalize.bind(null, this._options.cwd);
 
         this._files = Object.create(null);
@@ -155,6 +152,11 @@ class Processor {
     // Return the corrected-path version of the file
     normalize(file) {
         return this._normalize(file);
+    }
+
+    // Resolve a file from a src using the configured resolvers
+    resolve(src, file) {
+        return this._resolve(src, file);
     }
 
     // Check if a file exists in the currently-processed set

--- a/packages/processor/test/api.test.js
+++ b/packages/processor/test/api.test.js
@@ -336,7 +336,7 @@ describe("/processor.js", () => {
             });
         });
 
-        describe("._resolve()", () => {
+        describe(".resolve()", () => {
             it("should run resolvers until a match is found", () => {
                 let ran = false;
 
@@ -350,7 +350,7 @@ describe("/processor.js", () => {
                 
                 expect(
                     relative([
-                        processor._resolve(
+                        processor.resolve(
                             require.resolve("./specimens/start.css"),
                             "./local.css"
                         ),
@@ -370,7 +370,7 @@ describe("/processor.js", () => {
                 
                 expect(
                     relative([
-                        processor._resolve(
+                        processor.resolve(
                             require.resolve("./specimens/start.css"),
                             "./local.css"
                         ),

--- a/packages/svelte/package.json
+++ b/packages/svelte/package.json
@@ -31,8 +31,7 @@
   "dependencies": {
     "@modular-css/processor": "file:../processor",
     "escape-string-regexp": "^2.0.0",
-    "is-url": "^1.2.4",
-    "resolve-from": "^5.0.0"
+    "is-url": "^1.2.4"
   },
   "peerDependencies": {
     "svelte": ">1"

--- a/packages/svelte/svelte.js
+++ b/packages/svelte/svelte.js
@@ -2,7 +2,6 @@
 
 const path = require("path");
 
-const resolve = require("resolve-from");
 const isUrl = require("is-url");
 const escape = require("escape-string-regexp");
 
@@ -143,7 +142,7 @@ module.exports = (config = false) => {
             // Assign to file for later usage in logging
             css = href;
 
-            const external = resolve(path.dirname(html), css);
+            const external = processor.resolve(html, css);
 
             log("extract <link>", external);
 

--- a/packages/svelte/test/specimens/link-resolving.html
+++ b/packages/svelte/test/specimens/link-resolving.html
@@ -1,0 +1,3 @@
+<link rel="stylesheet" href="./does-not-exist.css" />
+
+<div class="{css.fooga}">fooga</div>

--- a/packages/svelte/test/svelte.test.js
+++ b/packages/svelte/test/svelte.test.js
@@ -370,4 +370,30 @@ describe("/svelte.js", () => {
 
         expect(results.map((result) => result.toString())).toMatchSnapshot();
     });
+
+    it("should use modular-css's file resolver", async () => {
+        const processor = new Processor({
+            namer,
+            resolvers : [
+                // Force all paths to resolve to a different file
+                () => require.resolve("./specimens/simple.css"),
+            ],
+        });
+
+        const filename = require.resolve("./specimens/link-resolving.html");
+        const { preprocess } = plugin({
+            processor,
+        });
+
+        const processed = await svelte.preprocess(
+            fs.readFileSync(filename, "utf8"),
+            Object.assign({}, preprocess, { filename })
+        );
+
+        expect(processed.toString()).toMatchSnapshot();
+
+        const output = await processor.output();
+
+        expect(output.css).toMatchSnapshot();
+    });
 });

--- a/packages/www/src/api/usage-js.md
+++ b/packages/www/src/api/usage-js.md
@@ -225,3 +225,18 @@ Remove files from the `Processor` instance. Accepts a single file or array of fi
 
 Returns an array of file paths. Accepts a single file argument to get the dependencies for, will return entire dependency graph in order if argument is omitted.
 
+##### `.invalidate(file)`
+
+Marks a file as stale, if that file is re-added either directly or as a dependency it will be reloaded from disk instead of the Processor cache.
+
+##### `.has(file)`
+
+Checks if the Processor instance knows about a file.
+
+##### `.normalize(file)`
+
+Uses the built-in path normalization settings to normalize the case of a file path.
+
+##### `.resolve(src, file)`
+
+Resolves `file` from `src`, using any of the specified [`resolvers`](#resolvers).

